primitives/account/src/logs.rs

```diff
@@ -0,0 +1,364 @@
+                staker_address,
+                validator_address,
+                ..
+            } => {
+                staker_address == address
+                    || validator_address
+                        .as_ref()
+                        .map(|validator_address| validator_address == address)
+                        .unwrap_or(false)
+            }
+            Log::Stake {
+                staker_address,
+                validator_address,
+                ..
+            } => {
+                staker_address == address
+                    || validator_address
+                        .as_ref()
+                        .map(|validator_address| validator_address == address)
+                        .unwrap_or(false)
+            }
+            Log::UpdateStaker {
+                staker_address,
+                old_validator_address,
+                new_validator_address,
+            } => {
+                staker_address == address
+                    || old_validator_address
+                        .as_ref()
+                        .map(|old_address| old_address == address)
+                        .unwrap_or(false)
+                    || new_validator_address
+                        .as_ref()
+                        .map(|new_address| new_address == address)
+                        .unwrap_or(false)
+            }
+            Log::DeleteValidator {
+                validator_address,
+                reward_address,
+            } => validator_address == address || reward_address == address,
+            Log::Unstake {
+                staker_address,
+                validator_address,
+                ..
+            } => {
+                staker_address == address
+                    || validator_address
+                        .as_ref()
+                        .map(|validator_address| validator_address == address)
+                        .unwrap_or(false)
+            }
+            Log::PayoutReward { to, .. } => to == address,
+            Log::Park {
+                validator_address, ..
+            } => validator_address == address,
+            Log::Slash {
+                validator_address, ..
+            } => validator_address == address,
+            Log::RevertContract { contract_address } => contract_address == address,
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+#[cfg_attr(feature = "serde-derive", derive(serde::Serialize, serde::Deserialize))]
+pub struct TransactionLog {
+    #[serde(rename = "hash")]
+    pub tx_hash: Blake2bHash,
+    pub logs: Vec<Log>,
+}
+
+impl TransactionLog {
+    pub fn new(tx_hash: Blake2bHash, logs: Vec<Log>) -> Self {
+        Self { tx_hash, logs }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+#[cfg_attr(feature = "serde-derive", derive(serde::Serialize, serde::Deserialize))]
+// Renaming affects only the struct names and thus their tag, the "type" field.
+#[serde(rename_all = "kebab-case", tag = "type")]
```

Comment:
Shouldn't this be `camelCase`? Maybe we can change it here and avoid adding it for all of the members?

Reply:
Additionally, this rule only applies to direct members of the ebum, not _their_ members in turn. That's why the rename rule needs to be added to all members individually anyway.

Reply:
If you find a way to apply it to all children fields as well, please tell!

Reply:
Ok, yeah I see why it needs to be in the enum variants. But still it doesn't look consistent to me that this enum's tag would follow kebab case when everything else does camel case.